### PR TITLE
Add `RSpec/PendingWithoutReason` cop

### PR DESCRIPTION
**Pull request by r7kamura**
## Motivation

I think it is a bad practice to use `pending` or `skip` without giving any reason. So, how about discouraging such practices by default?

## About naming

Since the convention of calling "pending or skip" as "pending" seems to already exist as `RSpec/Pending`, I decided to name it `RSpec/PendingWithoutReason` here as well for consistency. Incidentally, other conventions of calling "pending or skip" as "skip" also seem to exist in `RSpec::Cop::RSpec::SkipOrPending#skipped_in_metadata?`.

Also, `RuboCop::Cop::RSpec::Pending` says "pending or skip", while this `RSpec::Cop::RSpec::SkipOrPending` says "skip or pending". I decided to use "pending or skip" in this case in preference to `RSpec/Pending`.

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [x] Added tests.
- [x] Updated documentation.
- [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

If you have created a new cop:

- [x] Added the new cop to `config/default.yml`.
- [x] The cop is configured as `Enabled: pending` in `config/default.yml`.
- [x] The cop is configured as `Enabled: true` in `.rubocop.yml`.
- [x] The cop documents examples of good and bad code.
- [x] The tests assert both that bad code is reported and that good code is not reported.
- [x] Set `VersionAdded: "<<next>>"` in `default/config.yml`.

### CODE REVIEW

**Comment by r7kamura** [comment]
Regardless of whether this Cop is added or not, the other three commits are changes that should be put in independently of each other, and this Cop is dependent on those changes. That is why I am adding 4 commits this time. If you have a policy that each should be separated into a Pull Request, please let me know so I can try to separate them.

File: config/default.yml [hunk 1]
```diff
@@ -182,6 +182,61 @@ RSpec/BeforeAfterAll:
   StyleGuide: https://rspec.rubystyle.guide/#avoid-hooks-with-context-scope
   Reference: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/BeforeAfterAll
 
+RSpec/Capybara:
```
Comment (pirj) [inline config/default.yml:185]: The reason for such ordering was to put departments below main cops

File: lib/rubocop/cop/rspec/pending_without_reason.rb [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      #   xdescribe 'something' do
+      #   end
+      #
+      #   # bad
+      #   skip 'does something' do
+      #   end
+      #
+      #   # bad
+      #   it 'does something', :skip do
+      #   end
```
Comment (pirj) [inline lib/rubocop/cop/rspec/pending_without_reason.rb:32]: How about `it 'works on newer Ruby versions', skip: RUBY_VERSION < 3.0` do?

File: spec/project/default_config_spec.rb [hunk 3]
```diff
@@ -47,7 +47,7 @@ def cop_configuration(config_key)
       .to match_array([*config_keys, 'Metrics/BlockLength'])
   end
 
-  it 'sorts configuration keys alphabetically', :pending do
+  it 'sorts configuration keys alphabetically' do
```
Comment (pirj) [inline spec/project/default_config_spec.rb:50]:
```suggestion
  it 'sorts configuration keys alphabetically', pending: 'sort putting sub-departments below' do
```

File: spec/rubocop/cop/rspec/pending_without_reason_spec.rb [hunk 4]
```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::PendingWithoutReason do
+  context 'when pending by pending step with reason' do
+    it 'registers no offense' do
```
Comment (pirj) [inline spec/rubocop/cop/rspec/pending_without_reason_spec.rb:5]: ❤️ for putting good examples above

**Review by pirj — APPROVED** [review]
Looks great, thank you!
I'd revert the reordering of cops unless there's a good reason.

File: spec/project/default_config_spec.rb (hunk 3 above)
Comment (bquorning) [reply spec/project/default_config_spec.rb:50]: I tried rewriting this spec in #1507. That PR could replace this commit.

File: lib/rubocop/cop/rspec/pending_without_reason.rb [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            #ExampleGroups.skipped
+          PATTERN
+        )
+
+        # @!method skipped_by_metadata_without_reason?(node)
+        def_node_matcher :skipped_by_metadata_without_reason?, <<~PATTERN
+          (send #rspec? {#ExampleGroups.all #Examples.all} ... {<(sym :skip) ...> (hash <(pair (sym :skip) true) ...>)})
+        PATTERN
+
+        def on_send(node)
```
Comment (bquorning) [inline lib/rubocop/cop/rspec/pending_without_reason.rb:90]: Can we use `RESTRICT_ON_SEND` to improve performance?

Comment (r7kamura) [reply lib/rubocop/cop/rspec/pending_without_reason.rb:90]: This cop needs to inspect user-defined methods for Example and ExampleGroup, so it is not possible to filter them with pre-defined `RESTRICT_ON_SEND` in advance.

File: lib/rubocop/cop/rspec/pending_without_reason.rb (hunk 2 above)
Comment (r7kamura) [reply lib/rubocop/cop/rspec/pending_without_reason.rb:32]: Ideally, we would like to make it an offense, but in this case the offense is not detectable. In the current implementation, it is only considered an offense when the metadata is specified in Symbol or Boolean literal.

**Review by bquorning — APPROVED** [review]
Do we need to cover edge cases like `skip: false` / `skip: nil` / `pending: false` / `pending: nil`? I think perhaps we don’t.

**Comment by pirj** [comment]
Thank you, @r7kamura !

**Pull request merged**